src/Microsoft.SymbolStore/SymbolStores/HttpSymbolStore.cs

```diff
@@ -121,7 +121,7 @@ protected override async Task<SymbolStoreFile> GetFileInner(SymbolStoreKey key,
 
         protected Uri GetRequestUri(string index)
         {
-            if (!Uri.TryCreate(Uri, index, out Uri requestUri))
```

Comment:
Are we sure `EscapeUriString` is the right API? This closes the number of unsupported files, as `EscapeDataString` would always escape `:/?#[]@!$&'()*+,;=`. Hence, this covers the space case, but on several ext file systems most of these are valid in the file names. These would need to be escaped on each part of the triplet I believe. However, this is such an obscure case that I'm ok with this part of the change.

Reply:
EscapeDataString escapes too much especially the "/" the parts of the index. The difference between doing the escaping here and in BuildKey is that this code is escaping the whole index and the BuildKey code was escaping only the file name portion.